Upper tx validity could be very big on close/contest (#615)
input-output-hk/hydra: CHANGELOG.md

  + Changed logs of `BeginInitialize` and `EndInitialize`.
  + Added `SkipUpdate` constructor to the wallet logs.

> - **BREAKING** Implemented [ADR21](https://hydra.family/head-protocol/adr/21) to introduce bounded close transactions:

Reviewer: The fact that this is ADR21 is a bit of a side information. It's much more important to users that we essentially changed the way the contestation deadline is computed (and added a new command line option).

  + There is a new hydra-node flag `--contestation-period` to control the close tx validity bounds as well as

>     determine the contestation deadline.

Reviewer: Can you explain what the node does to users? Before they could provide the contestation period on the `Init` input. Now this is different. The deadline is always `2 * --contestation-period` after the `Close`.

>   + If hydra-node receives a `Init` tx with the _wrong_ contestation period then this tx is ignored which implies

Reviewer: ```suggestion
  + If hydra-node receives a `Init` tx with a contestation period _not matching_ `--contestation-period`, this tx is ignored which implies
```

  + Will make bumping dependencies (e.g. cardano-node) easier.
  + Build commands change, see updated [Contribution Guidelines](https://github.com/input-output-hk/hydra/blob/master/CONTRIBUTING.md)

- **BREAKING** Change the way tx validity and contestation deadline is constructed for close transactions:
  + There is a new hydra-node flag `--contestation-period` expressed in seconds
    to control the close tx validity bounds as well as determine the

>     contestation deadline.

Reviewer: We should add an example: `e.g. with `--contestation-period 30s`, the node will close the head `30s` after submitting the close transaction and other parties will have another `30s` to contest.

And/or add a section in the user manual (where we describe also the fuel stuff e.g) about this.

>   + The deadline is always `2 * --contestation-period` after the Close tx.

Reviewer: this is not true anymore with `min(maxGraceTime, contestationPeriod)`. Instead it may be "up to 2*cp"
